libkbfs/tlf_journal.go

```diff
@@ -460,6 +479,10 @@ func (j *tlfJournal) flushOneBlockOp(ctx context.Context) (bool, error) {
 		return false, err
 	}
 
+	// TODO: There is a race condition if we shut down after we've
+	// flushed an op but before we remove it. Make sure we handle
+	// re-flushed ops idempotently.
```

Comment:
This seems pretty important.  Is there a JIRA issue you can cite here?  Presumably if everything is tagged with a revision number, you can check whether the server already has the revision or not.